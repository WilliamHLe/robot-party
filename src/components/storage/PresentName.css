.introductionWrapper {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}
.toggleButton {
    box-shadow: 0 4px 8px 0 #dcdae3;
    background-color: #0D1A2B;
    font-size: 20px;
    color: white;
    cursor: pointer;
    padding: 10px;
    width: 30%;
    text-align: left;
    border: none;
    border-radius: 6px;
    display: inline-block;
}

.toggleButton:hover {
    background-color: #1c375c;
}


.plusMinus {
    float: right;
    font-size: 20px;
    font-weight: bold;
}
