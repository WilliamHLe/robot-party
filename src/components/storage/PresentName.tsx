import React, {useState} from "react";
import {getJson} from "./JsonStorage";
import ChangeName from "./ChangeName";
import "./PresentName.css";

//Komponent som viser introduksjonen øverst på siden som sier enten "Hei!" eller "Hei, Navn!"
//Henter et navn til introduksjonen dersom det fins i localstorage
const PresentName = () => {

    let personalInfo = getJson("personalInfo") || "{}";

    const [visible, setVisible] = useState<boolean>(false);
    const [icon, setIcon] = useState<string>("+")

    let introduction:string;

    if ("name" in personalInfo) {
        introduction = "Hei, " + personalInfo.name + "!"
    }
    else {
        introduction = "Hei!"
    }

    const toggleChange = () => {
        setVisible(!visible) //toggler mellom false og true hver gang man trykker på knappen

        if (visible) {
            setIcon("+")
        }
        else {
            setIcon(("-"))
        }
    }

    return (
        <div className={"introductionWrapper"}>
            <h1>{introduction}</h1>
            <button onClick={toggleChange} className={"toggleButton"}>
                <span>Endre navn</span><span className={"plusMinus"}>{icon}</span>
            </button>
            {visible ? <ChangeName/> : null}
        </div>
    )
};

export default PresentName;