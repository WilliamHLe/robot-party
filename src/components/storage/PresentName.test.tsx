import React from 'react';
import PresentName from "./PresentName";
const renderer = require("react-test-renderer");


it('renders correctly', () => {
    const tree = renderer.create(<PresentName />).toJSON();
    expect(tree).toMatchSnapshot();
});

//test localstorage?