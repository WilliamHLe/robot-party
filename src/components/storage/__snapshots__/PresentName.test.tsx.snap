// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="introductionWrapper"
>
  <h1>
    Hei!
  </h1>
  <button
    className="toggleButton"
    onClick={[Function]}
  >
    <span>
      Endre navn
    </span>
    <span
      className="plusMinus"
    >
      +
    </span>
  </button>
</div>
`;
